keyring/backend.py

```diff
@@ -127,10 +127,9 @@ def get_credential(self, service, username):
         returned username.
         """
         # The default implementation requires a username here.
-        if username is not None:
-            password = self.get_password(service, username)
-            if password is not None:
-                return credentials.SimpleCredential(username, password)
+        cred = self.get_password(service, username)
```

Comment:
I don't think this code will work across all backends. The interface for `.get_password` is to return a text string. The whole point of `get_credential` was to allow backends to retrieve a credential for a service without supplying a username. See the WinVaultKeyring for a non-fallback implementation of `get_credential`. Perhaps that's the method you wish to override in the macOS and Linux backends.

Reply:
I agree, I was going about this the wrong way. Restored this back to original logic.